### PR TITLE
xSDK examples v0.4.0

### PR DESCRIPTION
**Pull request by v-dobrev**
New release, v0.4.0, of the `xsdk-examples` package.

### CODE REVIEW

File: var/spack/repos/builtin/packages/xsdk-examples/package.py [hunk 1]
```diff
@@ -17,6 +17,8 @@ class XsdkExamples(CMakePackage, CudaPackage):
     maintainers("balay", "luszczek", "balos1", "shuds13", "v-dobrev")
 
     version("develop", branch="master")
+    # FIXME: replace the testing branch with the actual release when available:
+    version("0.4.0", branch="testing-v0.4.0")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/xsdk-examples/package.py]: Is this going to happen soon? If so, I'd prefer to wait a couple of days and have the sha256 here.

Comment (v-dobrev) [reply var/spack/repos/builtin/packages/xsdk-examples/package.py]: The release will happen around mid-May (that's why I added the `WIP` and `don't-merge-yet` labels). The plan is to merge this PR after that. For now, we're just gonna use this branch/PR for testing.

**Comment by v-dobrev** [comment]
@spackbot help

**Comment by v-dobrev** [comment]
@spackbot fix style

**Comment by v-dobrev** [comment]
@spackbot fix style

File: var/spack/repos/builtin/packages/xsdk-examples/package.py [hunk 2]
```diff
@@ -103,5 +103,5 @@ def enabled(pkg):
         return args
 
     def check(self):
-        with working_dir(self.build_directory):
+        with working_dir(self.builder.build_directory):
```
Comment (v-dobrev) [inline var/spack/repos/builtin/packages/xsdk-examples/package.py]: @alalazo,  FYI:

This looks like a general Spack issue:

After merging the latest `develop` branch in this PR, `self.build_directory` was no longer working with error message
```console
'XsdkExamples' object has no attribute 'build_directory'
```
I had to replace it with `self.builder.build_directory` to get it to work.

**Review by mgates3 — COMMENTED** [review]
The SLATE changes look good. I didn't look at other packages.

**Comment by v-dobrev** [comment]
This branch is now ready for final review and merging.

**Review by balos1 — APPROVED** [review]

**Review by ulrikeyang — APPROVED** [review]

**Review by luszczek — APPROVED** [review]

**Pull request merged**